virtcontainers/qemu.go

```diff
@@ -892,33 +893,27 @@ func (q *qemu) hotplugNetDevice(endpoint Endpoint, op operation) error {
 		endpoint.SetPciAddr(pciAddr)
 
 		devID := "virtio-" + tap.ID
-		if err = q.qmpMonitorCh.qmp.ExecuteNetPCIDeviceAdd(q.qmpMonitorCh.ctx, tap.Name, devID, endpoint.HardwareAddr(), addr, bridge.ID, romFile, int(q.config.NumVCPUs), q.arch.runNested()); err != nil {
-			return err
-		}
-	} else {
-		switch endpoint.Type() {
-		case VethEndpointType:
-			drive := endpoint.(*VethEndpoint)
-			tap = drive.NetPair.TapInterface
-		case TapEndpointType:
-			drive := endpoint.(*TapEndpoint)
-			tap = drive.TapInterface
-		default:
-			return fmt.Errorf("this endpoint is not supported")
-		}
 
-		if err := q.removeDeviceFromBridge(tap.ID); err != nil {
-			return err
+		if machine, _ := q.getQemuMachine(); machine.Type == QemuCCWVirtio {
```

Comment:
```go
	if machine, _ := q.getQemuMachine(); machine.Type == QemuCCWVirtio {
		return q.qmpMonitorCh.qmp.ExecuteNetCCWDeviceAdd(q.qmpMonitorCh.ctx, tap.Name, devID, endpoint.HardwareAddr(), addr, bridge.ID, int(q.config.NumVCPUs))
	}
	return q.qmpMonitorCh.qmp.ExecuteNetPCIDeviceAdd(q.qmpMonitorCh.ctx, tap.Name, devID, endpoint.HardwareAddr(), addr, bridge.ID, romFile, int(q.config.NumVCPUs), q.arch.runNested())
```

I think this looks better, @alicefr what do you think?

Reply:
I changed the code as you suggested

Reply:
@sboeuf you're right, I can add a check